apps/src/dance/Dance.js

```diff
@@ -628,12 +628,22 @@ Dance.prototype.execute = async function () {
   await this.initSongsPromise;
 
   const songMetadata = await this.songMetadataPromise;
-  return new Promise((resolve, reject) => {
-    this.nativeAPI.play(songMetadata, success => {
-      this.performanceData_.lastRunButtonDelay =
-        performance.now() - this.performanceData_.lastRunButtonClick;
-      success ? resolve() : reject();
+  const userBlockTypes = [];
+  Blockly.getMainWorkspace()
+    .getAllBlocks()
+    .forEach(block => {
+      userBlockTypes.push(block.type);
     });
```

Comment:
nit: you could use `.map` here?

Reply:
I'm sorry - I didn't see this till after I merged. I'll add this change to my next PR https://github.com/code-dot-org/code-dot-org/pull/54218